packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -0,0 +1,227 @@
+      deploymentInformer: this.createDeploymentInformer(kc, ns, context),
+    };
+  }
+
+  private createPodInformer(kc: KubeConfig, ns: string, context: KubeContext): Informer<V1Pod> & ObjectCache<V1Pod> {
+    const k8sApi = kc.makeApiClient(CoreV1Api);
+    const listFn = () => k8sApi.listNamespacedPod(ns);
+    const path = `/api/v1/namespaces/${ns}/pods`;
+    return this.createInformer<V1Pod>(kc, context, path, listFn, {
+      onAdd: _obj => this.safeSetState(context.name, state => state.podsCount++),
+      onDelete: _obj => this.safeSetState(context.name, state => state.podsCount--),
+      onReachable: reachable => this.safeSetState(context.name, state => (state.reachable = reachable)),
+    });
+  }
+
+  private createDeploymentInformer(
+    kc: KubeConfig,
+    ns: string,
+    context: KubeContext,
+  ): Informer<V1Deployment> & ObjectCache<V1Deployment> {
+    const k8sApi = kc.makeApiClient(AppsV1Api);
+    const listFn = () => k8sApi.listNamespacedDeployment(ns);
+    const path = `/apis/apps/v1/namespaces/${ns}/deployments`;
+    return this.createInformer<V1Deployment>(kc, context, path, listFn, {
+      onAdd: _obj => this.safeSetState(context.name, state => state.deploymentsCount++),
+      onDelete: _obj => this.safeSetState(context.name, state => state.deploymentsCount--),
+    });
+  }
+
+  private createInformer<T extends KubernetesObject>(
+    kc: KubeConfig,
+    context: KubeContext,
+    path: string,
+    listPromiseFn: ListPromise<T>,
+    options: CreateInformerOptions<T>,
+  ): Informer<T> & ObjectCache<T> {
+    const informer = makeInformer(kc, path, listPromiseFn);
+
+    informer.on('add', (obj: T) => {
+      options.onAdd?.(obj);
+      options.onReachable?.(true);
+    });
+
+    informer.on('delete', (obj: T) => {
+      options.onDelete?.(obj);
+      options.onReachable?.(true);
+    });
+    informer.on('error', (err: unknown) => {
+      if (err !== undefined) {
+        console.error(`informer error on path ${path} for context ${context.name}: `, String(err));
+      }
+      options.onReachable?.(err === undefined);
+      // Restart informer after 5sec
+      setTimeout(() => {
+        this.restartInformer<T>(informer, context, options);
+      }, 5000);
+    });
+
+    if (options.onReachable) {
+      informer.on('connect', (err: unknown) => {
+        if (err !== undefined) {
+          console.error(`informer connect error on path ${path} for context ${context.name}: `, String(err));
+        }
+        options.onReachable?.(err === undefined);
+      });
+    }
+    this.restartInformer<T>(informer, context, options);
+    return informer;
+  }
+
+  private restartInformer<T>(
+    informer: Informer<KubernetesObject> & ObjectCache<KubernetesObject>,
+    context: KubeContext,
+    options: CreateInformerOptions<T>,
+  ) {
+    informer.start().catch((err: unknown) => {
+      if (err !== undefined) {
+        console.log('informer start error: ', String(err));
```

Comment:
```suggestion
        console.error('informer start error: ', String(err));
```
(or warn but probably not console.log)